src/core/3d/qgsabstract3drenderer.h

```diff
@@ -64,6 +65,27 @@ class CORE_EXPORT QgsAbstract3DRenderer SIP_ABSTRACT
     //! Resolves references to other objects - second phase of loading - after readXml()
     virtual void resolveReferences( const QgsProject &project );
 
+    /**
+     * Returns whether the current renderer could be updated from \a renderer
+     * \since QGIS 3.28
+     */
+    virtual bool supportsUpdateFrom( QgsAbstract3DRenderer *renderer ) const
+    {
+      Q_UNUSED( renderer );
+      return false;
+    }
+
+    /**
+     * Updates the current renderer and returns true in case the 3D scene needs to be refreshed
+     * \warning The renderer \a renderer must be of the same type with the current renderer
+     * \sinec QGIS 3.28
+     */
+    virtual bool updateCurrentRenderer( QgsAbstract3DRenderer *renderer )
+    {
+      Q_UNUSED( renderer );
+      return false;
+    };
+
```

Comment:
What about the other inherited classes (QgsMeshLayer3DRenderer, QgsVectorLayer3DRenderer and QgsRuleBased3DRenderer)?

Reply:
The same logic will work for the other renderers: when the changed setting is only one QParameter passed to the shaders, change that parameter instead of updating the whole scene